service/service.go

```diff
@@ -52,17 +73,64 @@ func (svc *Service) Start() {
 
 				message.Commit()
 				log.Debug("message committed", log.Data{"instance_id": instanceID})
-			case errorMessage := <-svc.Consumer.Errors():
-				log.Error(fmt.Errorf("aborting"), log.Data{"message_received": errorMessage})
-				svc.Consumer.Closer() <- true
-				svc.Producer.Closer() <- true
-				exitCh <- true
-				return
-			case <-healthCh:
+			case consumerError := <-svc.Consumer.Errors():
+				log.Error(fmt.Errorf("aborting consumer"), log.Data{"message_received": consumerError})
+				close(done)
+			case producerError := <-svc.Producer.Errors():
+				log.Error(fmt.Errorf("aborting producer"), log.Data{"message_received": producerError})
+				close(done)
+			case <-apiErrors:
+				log.Error(fmt.Errorf("server error forcing shutdown"), nil)
+				close(done)
 			}
 		}
 	}()
-	<-exitCh
 
-	log.Info("service dimension extractor stopped", nil)
+	select {
+	case <-done:
+		log.Debug("Quitting after done was closed", nil)
+	case <-signals:
```

Comment:
you're listening to a signal in two places, surely only one receiver will get that signal (it should be this one, because it's higher priority)
I think this block should then tell the goroutine (above) to exit